libs/blocks/gnav/gnav-contextual-search.js

```diff
@@ -53,25 +52,33 @@ function highlightTextElements(terms, elements) {
   });
 }
 
-async function fetchResults(signal, terms, config) {
-  let hits = [];
+async function getResults(signal, terms, config, resultsEl, dataSegment, prevHits) {
```

Comment:
I think this many params is a sign that this function can/should be broken up. Because of the number of checks based on input, the recursive pattern might not be the best. We could keep everything needed for triggering the abort signal enclosed within a function and adding a `while` loop I think. 

That being said, I am not sure fixing a bug is the right place to do a code overhaul like that. Do you think we ought to make a ticket for tech debt? Or do you like this pattern as is?

Reply:
It seems like you're describing how it was originally written, but that was causing the duplication bug https://github.com/adobecom/milo/blob/063d5b5dd99c6691c36ac560b7c936017ccb44d6/libs/blocks/gnav/gnav-contextual-search.js

I went through several patterns and this made the most sense to me. I'm open to suggestions though. Do you have a code sample?

Reply:
I think if you went through several and found this to make the most sense that is fine for me